src/blocks/menu/MenuItem.tsx

```diff
@@ -0,0 +1,67 @@
+import { FC } from 'react';
+import styled from 'styled-components';
+
+import { MenuItemComponentProps } from './Menu.types';
+import { ModeProp } from 'blocks/Blocks.types';
+import { getBlocksColor } from '../Blocks.utils';
+import { useBlocksTheme } from 'blocks/Blocks.hooks';
+import { Text } from 'blocks/text';
+import { Box } from 'blocks/box';
+
+const StyledMenuItem = styled.div.withConfig({
+  shouldForwardProp: (prop, defaultValidatorFn) => !['mode'].includes(prop) && defaultValidatorFn(prop),
+})<MenuItemComponentProps & ModeProp>`
+  // Menu default styles
+  padding: var(--s1);
+  display: flex;
+  flex-direction: row;
+  flex: 1;
+  align-items: center;
+  gap: var(--s1);
+  border-radius: var(--r2);
+  &:hover {
+    background-color: ${({ mode }) => getBlocksColor(mode, { light: 'gray-100', dark: 'gray-1000' })};
+  }
+  .menu-icon {
+    svg {
+      width: 100%;
+      height: 100%;
+    }
+  }
+  cursor: pointer;
+  /* Extra CSS props */
+  ${(props) => props.css || ''}
+`;
+
+const MenuItem: FC<MenuItemComponentProps> = ({ icon, label, onClick, ...props }) => {
+  const { mode } = useBlocksTheme();
+
+  return (
+    <StyledMenuItem
+      onClick={onClick}
+      mode={mode}
+      {...props}
+    >
+      {icon && (
+        <Box
+          width="24px"
+          height="24px"
```

Comment:
We should not pass this from here, since the height width of the icon can be controlled from outside so this has no effect on icon.